src/spatio_temporal_voxel_layer.cpp

```diff
@@ -62,15 +64,15 @@ void SpatioTemporalVoxelLayer::onInitialize(void)
 
   // initialize parameters, grid, and sub/pubs
   ros::NodeHandle nh("~/" + name_), g_nh, prefix_nh;
+  _nh = &nh;
```

Comment:
I don't understand the need for this (also its definition in the header needs to be tabbed over with the other variables. Why a pointer?)

Reply:
In order to create the publisher on the dynamic reconfigure callback, the node handler is needed. Either this object is created in the callback it self and then thrown away or the class has an instance of the node hadler that other methods can use. This instance can be either the object or a pointer to the object.

It is  a pointer because in the constructor is not meant to initialize anything in costmap\_2d and I think is a bit of waste create a full Node Handler to override it later on. All variables that are not meant to be initialized in the constructor but that need to be private members of the class I tend to create pointers so they can, later on, be filled. 

Ideally we would be utilizing the \_nh more than the nh and frieds, but as a prototype i didnt want to touch that part of the code until further discussed.

So far, I see three options:
1. NH in the callback
2. NH in class as object
3. NH in class as pointer

To me the cleanest is the third option

Reply:
Oh! I have just realized that I didnt use it a all! Well, the comment still makes sense :stuck_out_tongue: